core/src/apps/zcash/get_address.py

```diff
@@ -0,0 +1,81 @@
+from typing import TYPE_CHECKING
+
+from trezor import wire
+from trezor.crypto import base58
+from trezor.crypto.scripts import sha256_ripemd160
+from trezor.messages import ZcashAddress
+from trezor.ui.layouts import show_address
+
+from apps.bitcoin import keychain as t_keychain
+from apps.common import address_type
+from apps.common.coininfo import CoinInfo, by_name
+from apps.common.paths import HARDENED, address_n_to_str
+
+from .orchard.keychain import OrchardKeychain
+from .unified import Typecode, encode_address
+
+if TYPE_CHECKING:
+    from trezor.wire import Context
+    from trezor.messages import ZcashGetAddress
+
+
+def encode_p2pkh(raw_address: bytes, coin: CoinInfo) -> str:
+    return base58.encode_check(address_type.tobytes(coin.address_type) + raw_address)
+
+
+async def get_address(ctx: Context, msg: ZcashGetAddress) -> ZcashAddress:
+    if not (msg.t_address_n or msg.z_address_n):
+        raise wire.DataError("t-address or z-address path expected")
+
+    coin = by_name(msg.coin_name)
```

Comment:
ah, one thing I didn't notice before: this code is not actually showing the path warnings.

both `get_raw_transparent` and `get_raw_orchard` should contain code like this:
```
if msg.show_display:
    await paths.validate_path(ctx, keychain, address_n)
```

Reply:
99b3804b50666483f0a8271929545538dcb9df23